### PR TITLE
feat(web): fullscreen view for irc announce logs

### PR DESCRIPTION
**Pull request by martylukyy**
This PR implements a "fullscreen" view for the new irc announce logs feature.
I tried to make the logs line break at reaching the fullscreen view toggle button, but couldn't wrap my head around it.
If someone has an idea on how to make the lines break upon reaching the fullscreen button,
feel free to add it! Would love to see a solution for this.

Otherwise here are some pictures of this features:
Feel free to suggest idea on how to improve or if you would change something on the design!

### CODE REVIEW

File: web/src/screens/settings/Irc.tsx [hunk 1]
```diff
@@ -644,11 +667,15 @@ export const Events = ({ network, channel }: EventsProps) => {
   }, [settings]);
 
   return (
-    <div className="dark:bg-gray-800 rounded-lg shadow-lg p-1.5">
+    <div className={`dark:bg-gray-800 rounded-lg shadow-lg p-2 ${isFullscreen ? "fixed top-0 left-0 w-screen h-screen z-50" : ""}`}>
```
Comment (zze0s) [inline web/src/screens/settings/Irc.tsx:670]: Could use the `classes` helper function which will clean this up a bit. Same on line `672` and `678`.

**Review by zze0s — COMMENTED** [review]
Great feature marty!

**Comment by stacksmash76** [comment]
> If someone has an idea on how to make the lines break upon reaching the fullscreen button,
feel free to add it! Would love to see a solution for this.

```
div.flex
    div.flex.flex-grow
        ... lines
    div.flex
        ... button
```

**Comment by martylukyy** [comment]
> > If someone has an idea on how to make the lines break upon reaching the fullscreen button,
> > feel free to add it! Would love to see a solution for this.
> 
> ```
> div.flex
>     div.flex.flex-grow
>         ... lines
>     div.flex
>         ... button
> ```

Thanks for the suggestion but @fabricionaweb and me failed miserably 😂 
At this point we will call it "good enough" and maybe z wants to play with it - we will see 😄

**Comment by stacksmash76** [comment]
So everything went to shit after I left? Nice.

```diff
diff --git a/web/src/screens/settings/Irc.tsx b/web/src/screens/settings/Irc.tsx
index f3aaf13..cdd0bb4 100644
--- a/web/src/screens/settings/Irc.tsx
+++ b/web/src/screens/settings/Irc.tsx
@@ -662,41 +680,45 @@ export const Events = ({ network, channel }: EventsProps) => {
     <div
       className={classNames(
         "dark:bg-gray-800 rounded-lg shadow-lg p-2",
-        isFullscreen ? "fixed top-0 left-0 w-screen h-screen z-50" : ""
+        isFullscreen ? "fixed top-0 left-0 right-0 bottom-0 w-screen h-screen z-50" : ""
       )}
     >
-      <div className="flex relative">
+      <div className="flex flex-row max-h-screen overflow-y-auto">
+        <div
+          className={classNames(
+            "overflow-auto mb-4 rounded-lg bg-gray-100 dark:bg-gray-900",
+            isFullscreen ? "p-2 border-4 border-gray-300 dark:border-gray-700" : "px-2 py-1 aspect-[2/1]"
+          )}
+        >
+          {logs.map((entry, idx) => (
+            <div
+              key={idx}
+              className={classNames(
+                settings.indentLogLines ? "grid justify-start grid-flow-col" : "",
+                settings.hideWrappedText ? "truncate hover:text-ellipsis hover:whitespace-normal" : ""
+              )}
+            >
+              <span className="mr-1 text-gray-700 dark:text-gray-500">
+                <span className="font-bold text-gray-500 dark:text-gray-600" title={simplifyDate(entry.time)}>{entry.nick}: </span>
+                <span className="font-mono">{entry.msg}</span>
+              </span>
+            </div>
+          ))}
+        </div>
+        <div className="display-none" ref={messagesEndRef} />
         <button
           className={classNames(
-            "dark:bg-gray-800 p-2 absolute top-2 right-2 mr-2 bg-gray-200 hover:bg-gray-300 dark:hover:bg-gray-700 hover:cursor-pointer rounded-md"
+            "h-fit max-h-full p-2 ml-2 mb-2 bg-gray-200 rounded-md hover:bg-gray-300 dark:bg-gray-800 dark:hover:bg-gray-700 hover:cursor-pointer"
           )}
           onClick={toggleFullscreen}
         >
-          {isFullscreen
-            ? <span className="flex items-center"><ArrowsPointingInIcon className="w-5 h-5"/></span>
-            : <span className="flex items-center"><ArrowsPointingOutIcon className="w-5 h-5"/></span>}
+          <span className="flex items-center">
+            {isFullscreen
+              ? <ArrowsPointingInIcon className="w-5 h-5"/>
+              : <ArrowsPointingOutIcon className="w-5 h-5"/>}
+          </span>
         </button>
       </div>
-      <div
-        className={classNames(
-          "overflow-y-auto rounded-lg min-w-full bg-gray-100 dark:bg-gray-900 overflow-auto",
-          isFullscreen ? "max-w-full h-full p-2 border-gray-300 dark:border-gray-700" : "px-2 py-1 aspect-[2/1]"
-        )}
-      >
-        {logs.map((entry, idx) => (
-          <div
-            key={idx}
-            className={classNames(
-              settings.indentLogLines ? "grid justify-start grid-flow-col" : "",
-              settings.hideWrappedText ? "truncate hover:text-ellipsis hover:whitespace-normal" : ""
-            )}
-          >
-            <span className="font-mono text-gray-500 dark:text-gray-500 mr-1"><span className="dark:text-gray-600" title={simplifyDate(entry.time)}>{entry.nick}:</span> {entry.msg}</span>
-          </div>
-        ))}
-        <div className="mt-6" ref={messagesEndRef} />
-      </div>
-
       {/*<div>*/}
       {/*  <form onSubmit={handleSubmit(onSubmit)}>*/}
       {/*    <input*/}
```

PS. btoa is deprecated
PS2. https://github.com/autobrr/autobrr/blob/develop/web/src/screens/settings/Releases.tsx#L101 `shadow-sm` creates an artifact here on the light theme
PS3. The colors are shite.

**Comment by KyleSanderson** [comment]
> So everything went to shit after I left? Nice.

That doesn't look like an offline modal for the PWA.

**Comment by stacksmash76** [comment]
> > So everything went to shit after I left? Nice.
> 
> That doesn't look like an offline modal for the PWA.

Why would it? I very much prefer stakes over sardines.

**Comment by s0up4200** [comment]
> > > So everything went to shit after I left? Nice.
> > 
> > 
> > That doesn't look like an offline modal for the PWA.
> 
> Why would it? I very much prefer stakes over sardines.

When seagulls follow the trawler, it is because they think sardines will be thrown into the sea.

**Comment by stacksmash76** [comment]
> When seagulls follow the trawler, it is because they think sardines will be thrown into the sea.

When you think about it... you let a cow grow, then you wait until it gets big and then you butcher it for the meat. Eventually the atoms which composed the cow end up in the sea and people and other creatures get to experience enjoyment over what beings enjoyed before them in different ways and forms. So, if you like sardines -- happy meal and enjoy mate!

**Review by zze0s — APPROVED** [review]

**Pull request merged**